components/bluetooth/bluetooth.c

```diff
@@ -217,12 +246,110 @@ void bt_app_gap_start_up(void)
 	
 	/* inititialize device information and status */
 	bt_app_gap_init();
+    
 	    
-    while (true) {
-	    ESP_LOGI(CSHA_TAG, "Begin inquiry");
-	    // esp_bt_gap_start_discovery(ESP_BT_INQ_MODE_GENERAL_INQUIRY, 10, 0);
+    for (;;)
+    {
+	    ESP_LOGI(BT_TAG, "Begin inquiry");
 	    esp_err_t discovery_result = esp_bt_gap_start_discovery(ESP_BT_INQ_MODE_GENERAL_INQUIRY, 10, 0);
-        ESP_LOGI(CSHA_TAG, "discovery result = %d", discovery_result);
+        ESP_LOGI(BT_TAG, "discovery result = %d", discovery_result);
 	    vTaskDelay(15000 / portTICK_PERIOD_MS);
     }
 }
+
+void ble_app_gap_cb(esp_gap_ble_cb_event_t event, esp_ble_gap_cb_param_t *param)
+{
+    esp_err_t err;
+    uint8_t* adv_name = NULL;
+    uint8_t adv_name_len = 0;
+    time_t now;
+    time(&now);
+    switch (event)
+    {
+        case ESP_GAP_BLE_SCAN_TIMEOUT_EVT:
+        case ESP_GAP_BLE_SCAN_PARAM_SET_COMPLETE_EVT:
+            esp_ble_gap_start_scanning(0);
+            break;
+        case ESP_GAP_BLE_SCAN_START_COMPLETE_EVT: 
+            if((err = param->scan_start_cmpl.status) != ESP_BT_STATUS_SUCCESS) 
+            {
+                ESP_LOGE(BLE_TAG,"Scan start failed: %s", esp_err_to_name(err));
+            }
+            else 
+            {
+                ESP_LOGI(BLE_TAG,"Start scanning...");
+            }
+            break;
+        case ESP_GAP_BLE_SCAN_RESULT_EVT:
+            switch(param->scan_rst.ble_evt_type)
+            {
+            case ESP_BLE_EVT_NON_CONN_ADV:
+            case ESP_BLE_EVT_SCAN_RSP:
+            case ESP_BLE_EVT_CONN_ADV:
+            case ESP_BLE_EVT_CONN_DIR_ADV:
+            case ESP_BLE_EVT_DISC_ADV: {
```

Comment:
Do you mean to have all of these cases fall through to `ESP_BLE_EVT_DISC_ADV` behavior?